### PR TITLE
LPD-36089 Fix broken aria reference

### PR DESCRIPTION
**Pull request by igor-franca**
Related [Ticket](https://liferay.atlassian.net/browse/LPD-36089)

If the captcha has a label, we will have a return other than false from the [getFieldDetails](https://github.com/liferay/liferay-portal/blob/e08d8ed42c872b7e16adc38ce1ff7ccbccce9662/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js#L59) function.

Which will trigger further actions that will ultimately make the validation of `hasFieldDetails` true [here](https://github.com/liferay/liferay-portal/blob/e08d8ed42c872b7e16adc38ce1ff7ccbccce9662/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js#L764-L772), thus constructing the span screen reader only element with the required reference containing the label.

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormDisplayContextUtil.java [hunk 1]
```diff
@@ -40,6 +42,16 @@ public static void addCaptchaDDMFormField(
 
 		captchaDDMFormField.setDataType("string");
 
+		LocalizedValue localizedValue = new LocalizedValue(
+			LocaleUtil.getDefault());
+
+		localizedValue.addString(
+			LocaleUtil.getDefault(), DDMFormFieldTypeConstants.CAPTCHA);
+
+		captchaDDMFormField.setLabel(localizedValue);
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormDisplayContextUtil.java:51]: @igor-franca I would simplify it to:

```
captchaDDMFormField.setDataType("string");
captchaDDMFormField.setLabel(
	new LocalizedValue() {
		{
			addString(
				LocaleUtil.getDefault(),
				DDMFormFieldTypeConstants.CAPTCHA);
		}
	});
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormDisplayContextUtil.java [hunk 2]
```diff
@@ -40,6 +42,16 @@ public static void addCaptchaDDMFormField(
...
+
+		captchaDDMFormField.setShowLabel(false);
+
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormDisplayContextUtil.java:54]: @igor-franca please move the 
```
captchaDDMFormField.setShowLabel(false);
```` 
to be after the 

```
captchaDDMFormField.setProperty("url", captchaResourceURL);
``` 
to follow alphabetical order.

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by igor-franca** [comment]
Thanks for the review @carolmariaabb, much better! See my changes [here](https://github.com/liferay-forms/liferay-portal/compare/9da2b00140efe4c843d11c239236448b4c092663..181ad53ef167113006737cd880c16558e5c4a46d)

**Comment by igor-franca** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]
LGTM 🎉

**Comment by igor-franca** [comment]
ci:test:playwright

**Comment by igor-franca** [comment]
ci:test:relevant

**Comment by igor-franca** [comment]
ci:test:forms

**Comment by igor-franca** [comment]
ci:test:data-engine

**Comment by igor-franca** [comment]
Changing the portlet url navigation pattern for some reason modified the behavior of an existing test. 
The commit responsible for this change has been removed.

cc: @cgmonte

**Comment by igor-franca** [comment]
ci:test:relevant

**Comment by igor-franca** [comment]
ci:test:stable

**Comment by igor-franca** [comment]
ci:forward

**Pull request closed without merging**